[WIP] Switch tests to gcc10 binaries (to check some obscure issue on CI)

azat opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

I found some obscure issue in gcc10 binaries, so let's force CI to check
them too (in the mean time), so:
- use gcc10 for various tests
- use gcc10 for address/thread/undefined sanitizers
- *note: fasttest leaved with clang, to avoid requiring force-test label in case of failure*

azat committed:
[WIP] Switch tests to gcc10 binaries (to check some obscure issue on CI)

I found some obscure issue in gcc10 binaries, so let's force CI to check
them too (in the mean time), so:
- use gcc10 for various tests
- use gcc10 for address/thread/undefined sanitizers

azat commented:
Does checks will be continued w/o sanitizers build success?

azat committed:
Use clang for sanitizers build

alexey-milovidov commented:
@azat I see strange fail of the `01581_to_int256_uint256_inf_nan` test.

azat commented:
>@azat I see strange fail of the 01581_to_int256_uint256_inf_nan test.

Yep, I saw it too, will take a look later.

My initial issues was in a different place, that was librdkafka, I already understand the culprit, but it requires changes in 2 other external repos (since it is actually not librkafka itself), so will get back once I will finish with this.

azat commented:
>@azat I see strange fail of the 01581_to_int256_uint256_inf_nan test.

@alexey-milovidov Actually looks like it is strange that it works in clang, since representing not finite numbers in numeric is UB

- https://godbolt.org/z/MYvGKq
- https://github.com/abseil/abseil-cpp/blob/1918ad2ae38aa32c74b558b322479a8efdd76363/absl/numeric/int128.cc#L248-L249

For this particular function it is easy to fix, the question is, is there any assumptions in the code that this should be defined?
Also looks like adding stateless tests under UBSAN is easy to add? Then it will catch such things automatically.

alexey-milovidov commented:
> Actually looks like it is strange that it works in clang, since representing not finite numbers in numeric is UB

Let's ask @myrrc or @ucasFL who are the authors of some changes around this code.

> Also looks like adding stateless tests under UBSAN is easy to add? Then it will catch such things automatically.

We already have:

> Functional stateless tests (ubsan) — fail: 0, passed: 2499, skipped: 9

azat commented:
>We already have:

Oh, missed that (I though there are already, but while searching on the page missed them).
So looks like it generates different code (due to non const value?) and in generated code there is no UB

azat commented:
https://clickhouse-builds.s3.yandex.net/17977/02b81b551b8b9171eeb68beb0d9591dd27121cc8/clickhouse_build_check/build_log_843372693_1607628818.txt

```
2020-12-10 18:31:33 /build/contrib/libcxx/include/__config:1098:25: error: expected unqualified-id before string constant
2020-12-10 18:31:33  1098 | _LIBCPP_FUNC_VIS extern "C" void __sanitizer_annotate_contiguous_container(
```

Interesting, gcc supports `extern "C"` only at the beginning of the line, hence it fails - https://github.com/ClickHouse-Extras/libcxx/pull/4

azat commented:
#### tsan

```
mv: error writing '/output/clickhouse-common-static_20.13.1.5425+tsan_amd64.deb': No space left on device
mv: error writing '/output/clickhouse-server_20.13.1.5425+tsan_all.deb': No space left on device
mv: error writing '/output/clickhouse-test_20.13.1.5425+tsan_all.deb': No space left on device
```

#### asan

```
2020-12-15 21:53:13 {standard input}: Assembler messages:
2020-12-15 21:53:13 {standard input}:7794067: Warning: end of file not at end of a line; newline inserted
2020-12-15 21:53:13 {standard input}:7794495: Error: unknown pseudo-op: `.lo'
2020-12-15 21:53:13 g++-10: fatal error: Killed signal terminated program cc1plus
```

Due to RSS limit I guess

#### ubsan

```
2020-12-15 21:40:16 {standard input}: Assembler messages:
2020-12-15 21:40:16 {standard input}:5216911: Warning: end of file not at end of a line; newline inserted
2020-12-15 21:40:16 {standard input}:5217054: Error: missing or invalid immediate expression `'
2020-12-15 21:40:16 {standard input}: Error: open CFI at the end of file; missing .cfi_endproc directive
2020-12-15 21:40:16 g++-10: fatal error: Killed signal terminated program cc1plus
```

Same

azat commented:
@alesapin @alexey-milovidov do you mind if I will try to increase rss limit (`ccache prlimit --rss=10000000 --cpu=600`) ?

azat commented:
@alesapin can you please take a look at the failures for 7d297b92a1a78914b7481c0e31dc34b17ec12dba (maybe there are some internal logs) ?
I doubt that this will be fixed by restarting them again (Even though I'm trying in the meantime).

azat committed:
Revert "Use clang for sanitizers build"

This reverts commit cf04d63f64809439e4b87bc7302357c3714f4961.

azat commented:
>ClickHouse build check — 12/14 builds are OK

```
2020-12-19 00:18:06 {standard input}:7690485: Warning: end of file not at end of a line; newline inserted
2020-12-19 00:18:06 {standard input}:7691756: Error: expected comma after name `_ZNK2DB25FunctionConvertFromStringINS_14DataTypeNumberIlEENS_8NameCastELNS_30ConvertFromStringExceptionModeE2ELNS_28ConvertFromStringParsingModeE0EE11executeImplERKNSt3' in .size directive
2020-12-19 00:18:06 g++-10: fatal error: Killed signal terminated program cc1plus
2020-12-19 00:18:06 compilation terminated.
```

Seems that it is still not enough resources

azat committed:
Add -g for release builds

azat commented:
>Stress test (thread) — Sanitizer assert (in stderr.log)

gcc's TSAN reports data race in `ColumnVector::insertRangeFrom`, can't reproduce for now

<details>

```
WARNING: ThreadSanitizer: data race (pid=284)
  Write of size 8 at 0x7fd9a10ad800 by thread T141:
    #0 _mm_store_si128 /usr/lib/gcc/x86_64-linux-gnu/10/include/emmintrin.h:721 (clickhouse+0x170d0024)
    #1 memcpy_fast ../contrib/FastMemcpy/FastMemcpy.h:630 (clickhouse+0x170d0024)
    #2 memcpy ../contrib/FastMemcpy/memcpy_wrapper.c:5 (clickhouse+0x170d0024)
    #3 memcpy ../contrib/libc-headers/x86_64-linux-gnu/bits/string_fortified.h:34 (clickhouse+0x1259c162)
    #4 DB::ColumnVector<unsigned long>::insertRangeFrom(DB::IColumn const&, unsigned long, unsigned long) ../src/Columns/ColumnVector.cpp:353 (clickhouse+0x1259c162)
    #5 DB::Set::insertFromBlock(DB::Block const&) ../src/Interpreters/Set.cpp:212 (clickhouse+0x12202c90)
    #6 DB::SelectQueryExpressionAnalyzer::tryMakeSetForIndexFromSubquery(std::__1::shared_ptr<DB::IAST> const&) ../src/Interpreters/ExpressionAnalyzer.cpp:323 (clickhouse+0x11e4c8ca)
    #7 DB::SelectQueryExpressionAnalyzer::makeSetsForIndex(std::__1::shared_ptr<DB::IAST> const&) ../src/Interpreters/ExpressionAnalyzer.cpp:379 (clickhouse+0x11e4d75c)
    #8 DB::ExpressionAnalysisResult::ExpressionAnalysisResult(DB::SelectQueryExpressionAnalyzer&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, bool, bool, bool, std::__1::shared_ptr<DB::FilterInfo> const&, DB::Block const&) ../src/Interpreters/ExpressionAnalyzer.cpp:1111 (clickhouse+0x11e51d2e)
    #9 DB::InterpreterSelectQuery::getSampleBlockImpl() ../src/Interpreters/InterpreterSelectQuery.cpp:521 (clickhouse+0x11dabb9d)
    #10 operator() ../src/Interpreters/InterpreterSelectQuery.cpp:406 (clickhouse+0x11daf95d)
    #11 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:409 (clickhouse+0x11dc2bee)
    #12 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #13 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #14 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #15 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #16 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #17 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #18 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #19 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #20 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #21 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #22 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #23 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #24 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #25 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #26 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #27 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #28 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #29 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #30 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #31 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #32 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #33 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #34 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #35 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #36 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #37 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #38 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #39 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #40 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #41 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #42 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #43 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #44 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #45 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #46 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #47 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #48 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #49 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #50 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #51 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #52 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #53 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #54 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #55 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #56 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #57 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #58 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #59 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #60 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #61 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #62 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #63 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #64 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #65 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #66 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #67 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #68 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #69 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #70 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #71 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #72 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #73 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #74 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #75 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #76 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #77 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #78 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #79 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x12111ffd)
    #80 DB::JoinedTables::makeLeftTableSubquery(DB::SelectQueryOptions const&) ../src/Interpreters/JoinedTables.cpp:153 (clickhouse+0x12111ffd)
    #81 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) ../src/Interpreters/InterpreterSelectQuery.cpp:290 (clickhouse+0x11dc3a38)
    #82 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectQuery.cpp:150 (clickhouse+0x11dc5be2)
    #83 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&>(std::__1::shared_ptr<DB::IAST> const&, DB::Context&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x1209824b)
    #84 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:327 (clickhouse+0x1209824b)
    #85 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) ../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:249 (clickhouse+0x12095691)
    #86 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../contrib/libcxx/include/memory:3028 (clickhouse+0x11cf358b)
    #87 DB::InterpreterFactory::get(std::__1::shared_ptr<DB::IAST>&, DB::Context&, DB::SelectQueryOptions const&) ../src/Interpreters/InterpreterFactory.cpp:110 (clickhouse+0x11cf358b)
    #88 executeQueryImpl ../src/Interpreters/executeQuery.cpp:503 (clickhouse+0x12398e13)
    #89 DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) ../src/Interpreters/executeQuery.cpp:869 (clickhouse+0x1239d031)
    #90 DB::TCPHandler::runImpl() ../src/Server/TCPHandler.cpp:260 (clickhouse+0x130bc327)
    #91 DB::TCPHandler::run() ../src/Server/TCPHandler.cpp:1414 (clickhouse+0x130bdc9a)
    #92 Poco::Net::TCPServerConnection::start() ../contrib/poco/Net/src/TCPServerConnection.cpp:43 (clickhouse+0x17186741)
    #93 Poco::Net::TCPServerDispatcher::run() ../contrib/poco/Net/src/TCPServerDispatcher.cpp:112 (clickhouse+0x171874ef)
    #94 Poco::PooledThread::run() ../contrib/poco/Foundation/src/ThreadPool.cpp:199 (clickhouse+0x1739c9b7)
    #95 run ../contrib/poco/Foundation/src/Thread.cpp:55 (clickhouse+0x1739697e)
    #96 Poco::ThreadImpl::runnableEntry(void*) ../contrib/poco/Foundation/src/Thread_POSIX.cpp:345 (clickhouse+0x17396c65)

  Previous write of size 1 at 0x7fd9a10ad800 by thread T1633:
    [failed to restore the stack]

  Thread T141 'TCPHandler' (tid=1178, running) created by thread T46 at:
    #0 pthread_create <null> (clickhouse+0x19756ca9)
    #1 Poco::ThreadImpl::startImpl(Poco::SharedPtr<Poco::Runnable, Poco::ReferenceCounter, Poco::ReleasePolicy<Poco::Runnable> >) ../contrib/poco/Foundation/src/Thread_POSIX.cpp:202 (clickhouse+0x17397a81)
    #2 Poco::Thread::start(Poco::Runnable&) ../contrib/poco/Foundation/src/Thread.cpp:128 (clickhouse+0x17398b58)
    #3 Poco::PooledThread::start() ../contrib/poco/Foundation/src/ThreadPool.cpp:85 (clickhouse+0x1739f91b)
    #4 Poco::ThreadPool::getThread() ../contrib/poco/Foundation/src/ThreadPool.cpp:461 (clickhouse+0x1739f91b)
    #5 Poco::ThreadPool::startWithPriority(Poco::Thread::Priority, Poco::Runnable&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) ../contrib/poco/Foundation/src/ThreadPool.cpp:365 (clickhouse+0x1739ffdc)
    #6 Poco::Net::TCPServerDispatcher::enqueue(Poco::Net::StreamSocket const&) ../contrib/poco/Net/src/TCPServerDispatcher.cpp:146 (clickhouse+0x1718730b)
    #7 Poco::Net::TCPServer::run() ../contrib/poco/Net/src/TCPServer.cpp:148 (clickhouse+0x17186154)
    #8 run ../contrib/poco/Foundation/src/Thread.cpp:55 (clickhouse+0x1739697e)
    #9 Poco::ThreadImpl::runnableEntry(void*) ../contrib/poco/Foundation/src/Thread_POSIX.cpp:345 (clickhouse+0x17396c65)

  Thread T1633 (tid=120048, finished) created by thread T200 at:
    #0 pthread_create <null> (clickhouse+0x19756ca9)
    #1 void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>) ../contrib/libcxx/include/__threading_support:394 (clickhouse+0x7d8738a)
    #2 ThreadPoolImpl<std::__1::thread>::scheduleOrThrow(std::__1::function<void ()>, int, unsigned long) ../src/Common/ThreadPool.cpp:153 (clickhouse+0x7d8ecce)
    #3 ThreadFromGlobalPool<DB::PipelineExecutor::executeImpl(size_t)::<lambda()> > ../src/Common/ThreadPool.h:162 (clickhouse+0x131cda78)
    #4 construct<ThreadFromGlobalPool, DB::PipelineExecutor::executeImpl(size_t)::<lambda()> > ../contrib/libcxx/include/memory:1876 (clickhouse+0x131cda78)
    #5 __construct<ThreadFromGlobalPool, DB::PipelineExecutor::executeImpl(size_t)::<lambda()> > ../contrib/libcxx/include/memory:1768 (clickhouse+0x131cda78)
    #6 construct<ThreadFromGlobalPool, DB::PipelineExecutor::executeImpl(size_t)::<lambda()> > ../contrib/libcxx/include/memory:1595 (clickhouse+0x131cda78)
    #7 __construct_one_at_end<DB::PipelineExecutor::executeImpl(size_t)::<lambda()> > ../contrib/libcxx/include/vector:924 (clickhouse+0x131cda78)
    #8 emplace_back<DB::PipelineExecutor::executeImpl(size_t)::<lambda()> > ../contrib/libcxx/include/vector:1683 (clickhouse+0x131cda78)
    #9 DB::PipelineExecutor::executeImpl(unsigned long) ../src/Processors/Executors/PipelineExecutor.cpp:747 (clickhouse+0x131cda78)
    #10 DB::PipelineExecutor::execute(unsigned long) ../src/Processors/Executors/PipelineExecutor.cpp:400 (clickhouse+0x131ce8ed)
    #11 threadFunction ../src/Processors/Executors/PullingAsyncPipelineExecutor.cpp:79 (clickhouse+0x131da75b)
    #12 operator() ../src/Processors/Executors/PullingAsyncPipelineExecutor.cpp:101 (clickhouse+0x131da75b)
    #13 __invoke_constexpr<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, uint64_t)::<lambda()>&> ../contrib/libcxx/include/type_traits:3525 (clickhouse+0x131da75b)
    #14 __apply_tuple_impl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, uint64_t)::<lambda()>&, std::__1::tuple<>&> ../contrib/libcxx/include/tuple:1415 (clickhouse+0x131da75b)
    #15 apply<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, uint64_t)::<lambda()>&, std::__1::tuple<>&> ../contrib/libcxx/include/tuple:1424 (clickhouse+0x131da75b)
    #16 operator() ../src/Common/ThreadPool.h:178 (clickhouse+0x131da75b)
    #17 __invoke<ThreadFromGlobalPool::ThreadFromGlobalPool<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, uint64_t)::<lambda()>, {}>::<lambda()>&> ../contrib/libcxx/include/type_traits:3519 (clickhouse+0x131daaaf)
    #18 __call<ThreadFromGlobalPool::ThreadFromGlobalPool<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, uint64_t)::<lambda()>, {}>::<lambda()>&> ../contrib/libcxx/include/__functional_base:348 (clickhouse+0x131daaaf)
    #19 operator() ../contrib/libcxx/include/functional:1540 (clickhouse+0x131daaaf)
    #20 operator() ../contrib/libcxx/include/functional:1714 (clickhouse+0x131daaaf)
    #21 std::__1::__function::__value_func<void ()>::operator()() const ../contrib/libcxx/include/functional:1867 (clickhouse+0x7d8c6f6)
    #22 std::__1::function<void ()>::operator()() const ../contrib/libcxx/include/functional:2473 (clickhouse+0x7d8c6f6)
    #23 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) ../src/Common/ThreadPool.cpp:236 (clickhouse+0x7d8c6f6)
    #24 ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}::operator()() const ../src/Common/ThreadPool.cpp:117 (clickhouse+0x7d8899e)
    #25 decltype (((forward<ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>)({parm#1}))()) std::__1::__invoke<ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>(ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}&&, (ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}&&)...) ../contrib/libcxx/include/type_traits:3519 (clickhouse+0x7d8899e)
    #26 void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>&, std::__1::__tuple_indices<>) ../contrib/libcxx/include/thread:273 (clickhouse+0x7d8899e)
    #27 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}> >(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>) ../contrib/libcxx/include/thread:284 (clickhouse+0x7d8899e)

SUMMARY: ThreadSanitizer: data race /usr/lib/gcc/x86_64-linux-gnu/10/include/emmintrin.h:721 in _mm_store_si128
```

</details>

azat committed:
Use release builds for sanitizers

This is an attempt to overcome an error (by not using debug symbols and -O3):

    2020-12-19 00:18:06 {standard input}:7690485: Warning: end of file not at end of a line; newline inserted
    2020-12-19 00:18:06 {standard input}:7691756: Error: expected comma after name `_ZNK2DB25FunctionConvertFromStringINS_14DataTypeNumberIlEENS_8NameCastELNS_30ConvertFromStringExceptionModeE2ELNS_28ConvertFromStringParsingModeE0EE11executeImplERKNSt3' in .size directive
    2020-12-19 00:18:06 g++-10: fatal error: Killed signal terminated program cc1plus
    2020-12-19 00:18:06 compilation terminated.

And this cannot be done in ci_config.json since this is hard-coded in
packager and sandbox.

azat committed:
Disable heavy build checker

azat committed:
Revert "Add -g for release builds"

This reverts commit bdf4c2213580ce3e9441e3cd50c0ad7c6fb79d0c.

azat committed:
Merge remote-tracking branch 'upstream/master' into gcc10

azat commented:
>Functional stateful tests (ubsan) — No output log

https://clickhouse-test-reports.s3.yandex.net/17977/ffa718d0c3a9d6166778b13e0bd3623f3e5173be/functional_stateful_tests_(ubsan).html

Tons of

```
../src/Common/MemoryTracker.h:142:32: runtime error: load of null pointer of type 'uint64_t'
../src/Common/CurrentThread.h:83:43: runtime error: load of null pointer of type 'struct ThreadStatus *'
```

>Functional stateful tests (address) — fail: 1, passed: 0

https://clickhouse-test-reports.s3.yandex.net/17977/ffa718d0c3a9d6166778b13e0bd3623f3e5173be/functional_stateful_tests_(address).html#fail1

```
==253==ERROR: AddressSanitizer: use-after-poison on address 0x62500003c100 at pc 0x00003c6e63ca bp 0x7fc56a4ba570 sp 0x7fc56a4b9d18
...
0x62500003c100 is located 0 bytes inside of 8192-byte region [0x62500003c100,0x62500003e100)
allocated by thread T71 (AsyncUpdater) here:
    #0 0x3c764777 in __interceptor_malloc (/usr/bin/clickhouse+0x3c764777)
    #1 0x9de4a21 in Allocator<false, false>::allocNoTrack(unsigned long, unsigned long) ../src/Common/Allocator.h:219
```

Looks like unpoison here does not work - https://github.com/ClickHouse/ClickHouse/blob/aff724ea7d371d322589fba96a1214aacab78e59/src/Common/Arena.h#L151

And indeed there is no call to __asan_unpoison_memory_region in asm, but there are calls to it in other places

(And yes gcc supports it - 
https://github.com/gcc-mirror/gcc/blob/ae99b315ba5b9e1ccc221b3c45de323cbc574400/libsanitizer/include/sanitizer/asan_interface.h#L69)

>Functional stateful tests (thread) — fail: 2, passed: 54, skipped: 3

https://clickhouse-test-reports.s3.yandex.net/17977/ffa718d0c3a9d6166778b13e0bd3623f3e5173be/functional_stateful_tests_(thread).html#fail1

```
WARNING: ThreadSanitizer: data race (pid=256)
  Write of size 8 at 0x7fad82ec8000 by thread T122:
    #0 _mm_storeu_si128(long long __vector(2)*, long long __vector(2)) /usr/lib/gcc/x86_64-linux-gnu/10/include/emmintrin.h:727 (clickhouse+0x1212e4a6)
    #1 detail::memcpySmallAllowReadWriteOverflow15Impl(char*, char const*, long) ../src/Common/memcpySmall.h:36 (clickhouse+0x1212e4a6)
    #2 memcpySmallAllowReadWriteOverflow15(void*, void const*, unsigned long) ../src/Common/memcpySmall.h:51 (clickhouse+0x1212e4a6)
    #3 DB::ColumnArray::replicateString(DB::PODArray<unsigned long, 4096ul, Allocator<false, false>, 15ul, 16ul> const&) const ../src/Columns/ColumnArray.cpp:1039 (clickhouse+0x1212e4a6)


  Previous write of size 8 at 0x7fad82ec8000 by thread T177:
    [failed to restore the stack]
```

Looks like the previous problem (with ASAN).

alexey-milovidov commented:
BTW, I'm not interested in gcc build with sanitizers at all (it has worse support than clang).

azat commented:
>BTW, I'm not interested in gcc build with sanitizers at all (it has worse support than clang).

@alexey-milovidov yep, I assumed that.

But the purpose is not to replace clang santitizers with gcc, but to run sanitizers against code generated by gcc, to catch possible issues like #18053 (even though gcc is not used for official builds, one day clang may have similar optimizations)
